### ISSUE TITLE
client: Best Peer Selection for Snap Sync Stalls due to Peer Status Issue in Devp2p

### ISSUE THREAD

**Issue by scorbajio** [ethereumjs/ethereumjs-monorepo#3222]
When the client finishes synchronizing the cl skeleton chain, snap sync does not start as expected but rather falls into a defunct hanging state. From internal discussion, it seems like it's because of a peering status issue in devp2p since it currently doesn't update peer status, which causes issues in best peer selection:
```
[12-29|10:27:16] INFO ---------------------------------------------------------------
[12-29|10:27:16] INFO Synchronized cl (skeleton) at height=399690 hash=0xa723…f737 🎉
[12-29|10:27:16] INFO ---------------------------------------------------------------
[12-29|10:27:16] INFO ------------------------------------------------------------
[12-29|10:27:16] INFO Synchronized blockchain at height=399690 hash=0xa723…f737 🎉
[12-29|10:27:16] INFO ------------------------------------------------------------
[12-29|10:27:24] INFO 
[12-29|10:27:24] INFO [ CL ] New consensus payload received  number=399690 hash=0xa723…f737 parentHash=0x58ce…2ba3  status=ACCEPTED gasUsed=0 baseFee=7 txs=0 withdrawals=0 
[12-29|10:27:24] INFO [ CL ] New chain head set (forkchoice update) head=0xa723…f737 finalized=0xa125…28fa response=SYNCING 
[12-29|10:27:24] INFO 
[12-29|10:27:24] INFO [ EL ] SYNCED (snapsync target=undefined vm=299319 cl=el=399691 ) synchronized=true peers=5 
[12-29|10:27:24] INFO [ EL ] snapsync=?? 
[12-29|10:27:24] INFO [ EL ] vm=299319 hash=0x2feb…4c44 started=false 
```

**Comment by ScottyPoi** [comment]
If you provide instructions on replicating the error, I will look into this.

**Comment by scorbajio** [comment]
Fixed by #3354
